cloudscript/cloudscript/CloudLib.cs

```diff
@@ -0,0 +1,1336 @@
+                    || hdr.ToString().Contains("Location"))
+                {
+                    int index = hdr.ToString().IndexOf('=');
+                    statusUri = hdr.ToString().Substring(index + 1);
+                    Console.WriteLine($"Got the status uri to poll for: {statusUri}");
+                    return statusUri;
+                }
+            }
+            Console.WriteLine("Did not get Azure-Asyncoperation key in response headers. Following are response headers");
+            foreach (var hdr in response.Headers)
+            {
+                Console.WriteLine(hdr.ToString());
+            }
+            throw new Exception($"Status uri not specified in response headers");
+        }
+
+        /// <summary>
+        /// Get azure access token from Saas to make REST calls to Saas.
+        /// </summary>
+        /// <param name="tenantId">Tenant Id</param>
+        /// <param name="clientId">Client Id</param>
+        /// <param name="clientSecretKey">Client secret key</param>
+        /// <returns>Access token</returns>
+        public static async Task<string> GetAzureAccessTokenAsync(string tenantId, string clientId, string clientSecretKey)
+        {
+            Console.WriteLine($"Fetching token for tenant {tenantId} with clientId {clientId}");
+            var context = new AuthenticationContext("https://login.windows.net/" + tenantId);
+            ClientCredential clientCredential = new ClientCredential(clientId, clientSecretKey);
+            var tokenResponse = await context.AcquireTokenAsync("https://management.azure.com/", clientCredential);
+            var accessToken = tokenResponse.AccessToken;
+            Console.WriteLine($"Got token: {accessToken}");
+            return accessToken;
+        }
+
+        /// <summary>
+        /// Start a VM.
+        /// </summary>
+        /// <param name="deviceName">ASE device name</param>
+        /// <param name="saasResourceGroup">Saas Resource Group containing ASE device</param>
+        /// <param name="vmName">VM name</param>
+        /// <param name="edgeResourceGroup">Resource Group on device</param>
+        /// <returns>None</returns>
+        public void StartVM(string deviceName, string saasResourceGroup, string vmName, string edgeResourceGroup)
+        {
+            Console.WriteLine($"Starting the VM {vmName}, RG {edgeResourceGroup}");
+            string response1 = GetAseResource(deviceName, saasResourceGroup);
+            dynamic resp1 = JObject.Parse(response1);
+            String edgeSubId = resp1.properties.edgeProfile.subscription.subscriptionId;
+            string uri = $"/subscriptions/{SubscriptionId}/resourcegroups/{saasResourceGroup}/providers/Microsoft.AzureStack/linkedSubscriptions/{edgeSubId}/linkedResourcegroups/{edgeResourceGroup}/linkedProviders/Microsoft.Compute/virtualMachines/{vmName}/start";
+            IRestResponse responsePost = MakeRestCall(uri, Method.POST, "2020-06-01-preview");
```

Comment:
Don't we need to pass **linked-api-version**

Reply:
it seems to be working without passing linked-api-version. So, didn't add